docs/native-modules-setup.md

```diff
@@ -229,3 +229,82 @@ If you are working on an existing module that already has iOS and Android sample
+
+### Setup CI (continuous integration) pipeline for your module
+
+When done developing your module, it's good practice to setup CI pipeline with automated build and tests to avoid any future regressions. There are many services avialable for setting up CI pipeline. We'll use [GitHub Actions](https://docs.github.com/en/actions/getting-started-with-github-actions/about-github-actions) as example here since it doesn't require extra account setup if you are already hosting your code on GitHub, also the default vm image has all the tools we needed pre-installed.
+
+The vm images supported by GitHub Actions CI/CD can be found [here](https://github.com/actions/virtual-environments#github-actions-virtual-environments), check the pre-installed toos and compare them with [React Native Windows development dependencies](https://microsoft.github.io/react-native-windows/docs/rnw-dependencies), find the image that has all (or most) of the required tools installed.
+
+Next you need to create a YAML file for GitHub Actions, the basic steps are:
+- Checkout code and setup the environment
+```yaml
+    - uses: actions/checkout@v2
+      name: Checkout Code
+     
+    - name: Setup Node.js
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12.9.1'
+
+    - name: Setup MSBuild
+      uses: microsoft/setup-msbuild@v1.0.0
+      with:
+        vs-version: 16.5
+       
+    - name: Setup NuGet
+      uses: NuGet/setup-nuget@v1.0.2
+
+    - name: Check node modules cache
+      uses: actions/cache@v1
+      id: yarn-cache
+      with:
+        path: ./node_modules
+        key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-yarn-
+
+    - name: Install node modules
+      if: steps.yarn-cache.outputs.cache-hit != 'true'
+      run: yarn --pure-lockfile
+    
+    - name: yarn build
+      if: steps.yarn-cache.outputs.cache-hit == 'true'
+      run: |
+        yarn build
+        yarn tsc
+```
+- Build the project
+```yaml
+    - name: NuGet restore
+      run: nuget restore YourTestApp.sln
+
+    - name: Build x64 release
+      run: msbuild YourTestApp.sln /p:Configuration=Release /p:Platform=x64 -m
+```
+- Run tests
+```yaml
+    - name: Deploy
+      shell: powershell
+      run: |
+        cd example
+        Copy-Item -Path windows\x64\Release -Recurse -Destination windows\
+        npx react-native run-windows --arch x64 --release --no-build --no-packager
+
+    - name: Start Appium server
+      shell: powershell
+      run: Start-Process PowerShell -ArgumentList "yarn appium"
+      
+    - name: Run tests
+      run: yarn test:windows

```

Comment:
I worry about deviating from run-windows too much (e.g. this custom deploy step isn't what apps normally do).
I remember we discussed this on Tatiana's PR but I forget why we can't just use run-windows ?

Reply:
Yeah I agree we should just use run-windows if possible, I'll do some more testing on this...

Reply:
I got some weird "resource busy or locked" errors when doing run-windows on react-native-webview. But I just tried it in a standalone project it seems to be working fine, it's probably a WebView only issue. I've updated the doc to just use `npx react-native run-windows` as general guidance. 

Here's the WebView error I got if you are curious:
```
Failed to construct transformer : error : EBUSY: resource busy or locked, open 'D:\react-native-community\react-native-
webview\example\msbuild.ProjectImports.zip' [D:\react-native-community\react-native-webview\example\windows\WebViewWind
ows\WebViewWindows.vcxproj]
      at Object.openSync (fs.js:448:3)
      at Object.readFileSync (fs.js:350:35)
      at getContent (D:\react-native-community\react-native-webview\node_modules\jest-haste-map\build\worker.js:149:41)
      at Object.<anonymous> (D:\react-native-community\react-native-webview\node_modules\jest-haste-map\build\worker.js
  :199:9)
      at Generator.next (<anonymous>)
      at asyncGeneratorStep (D:\react-native-community\react-native-webview\node_modules\jest-haste-map\build\worker.js
  :78:24)
      at _next (D:\react-native-community\react-native-webview\node_modules\jest-haste-map\build\worker.js:98:9)
      at D:\react-native-community\react-native-webview\node_modules\jest-haste-map\build\worker.js:103:7
      at new Promise (<anonymous>)
      at Object.<anonymous> (D:\react-native-community\react-native-webview\node_modules\jest-haste-map\build\worker.js
  :95:12) {
EXEC : error : EBUSY: resource busy or locked, open 'D:\react-native-community\react-native-webview\example\msbuild.Pro
jectImports.zip' [D:\react-native-community\react-native-webview\example\windows\WebViewWindows\WebViewWindows.vcxproj]
      at Object.openSync (fs.js:448:3)
      at Object.readFileSync (fs.js:350:35)
      at getContent (D:\react-native-community\react-native-webview\node_modules\jest-haste-map\build\worker.js:149:41)
      at Object.<anonymous> (D:\react-native-community\react-native-webview\node_modules\jest-haste-map\build\worker.js
  :199:9)
      at Generator.next (<anonymous>)
      at asyncGeneratorStep (D:\react-native-community\react-native-webview\node_modules\jest-haste-map\build\worker.js
  :78:24)
      at _next (D:\react-native-community\react-native-webview\node_modules\jest-haste-map\build\worker.js:98:9)
      at D:\react-native-community\react-native-webview\node_modules\jest-haste-map\build\worker.js:103:7
      at new Promise (<anonymous>)
      at Object.<anonymous> (D:\react-native-community\react-native-webview\node_modules\jest-haste-map\build\worker.js
  :95:12)
    type: 'Error',
    errno: -4082,
    syscall: 'open',
    code: 'EBUSY',
    path: 'D:\\react-native-community\\react-native-webview\\example\\msbuild.ProjectImports.zip'
  }
```

Reply:
@kaiguo thanks for looking into it - this is a known issue that we've added a workaround for in the rnw repo.
https://github.com/dotnet/msbuild/issues/5383

The workaround is to add this blacklist entry to `metro.config.js`:
```js
  resolver: {
    // We need a custom resolveRequest right now since our integration tests use a "windesktop" platform thats specific to integration tests.
    resolveRequest: reactNativePlatformResolver({
      windesktop: 'react-native-windows',
      windows: 'react-native-windows',
    }),
    extraNodeModules: {
      // Redirect react-native-windows to this folder
      'react-native-windows': rnwPath,
    },
    blacklistRE: blacklist([
      // Avoid error EBUSY: resource busy or locked, open '...\vnext\msbuild.ProjectImports.zip' when building 'vnext\Microsoft.ReactNative.sln' with '/bl'
      /.*\.ProjectImports\.zip/,
    ]),
  },
```

Reply:
Yes that worked, I'll update WebView repo. Thanks!